quic.nimble

```diff
@@ -11,5 +11,4 @@ requires "nimcrypto >= 0.5.4 & < 0.6.0"
 requires "ngtcp2 >= 0.32.0 & < 0.33.0"
 requires "upraises >= 0.1.0 & < 0.2.0"
```

Comment:
this is another dep we don't use in low-level libs

Reply:
For upraises, I would argue we won't need it in a few months anyway (after 1.6 migration), so it might be easier to just remove it then

Reply:
it should be a search/replace operation, no? with the new `when (NimMajor, NimMinor) < (1, 4): ... else: ...` style, we get rid of the warning and we don't have to use weird/unusual constructs

also, for low-level libs we might want to keep 1.2-compat longer than end-user apps (no point breaking things that work)

Reply:
It's kinda painful for ie:
https://github.com/status-im/nim-quic/blob/e48bc4547f133ecc8738590d39fd609bdae49337/quic/listener.nim#L31-L33

or
https://github.com/status-im/nim-quic/blob/e48bc4547f133ecc8738590d39fd609bdae49337/quic/helpers/asyncloop.nim#L3

We didn't have many of thoses in libp2p, but here there are a few such places